src/Esprima/Ast/ArrowFunctionExpression.cs

```diff
@@ -4,21 +4,15 @@ namespace Esprima.Ast
 {
     public class ArrowFunctionExpression : Node, Expression, IFunction
     {
-        public bool Strict { get; }
-
         public Identifier Id { get; }
-
-        public List<INode> Params { get; }
-
+        public NodeList<INode> Params { get; }
```

Comment:
I don't know as my knowledge is still a bit lacking, could we have sort of ref-readonly solution

Reply:
I looked at the `JavaScript` parser and when the constructors are called with `Params` or `elements` it's once these collection are already built. Won't it just reallocate all its elements then only to have the collection as a struct? Where is the gain then?

Reply:
@lahma Yes, that could be done in a separate step and it would have to go on the interface! I just don't think that's something we need to address immediately in this PR. The situation is no different than with `List<>`.

@sebastienros There won't be any new allocation. During parsing, the collection/list is built up using `ArrayList<>`, which is just a thin (`struct`) wrapped around an array with an element count. When a node requiring such a list is constructed, the array ownership is _transferred_ from `ArrayList<>` to `NodeList<>`.

Reply:
> Where is the gain then?

For to answer the important bit. 😛So what we gain is that we can build a dynamic list over an array and expose the _same_ list/array publicly in a read-only manner without any additional allocation.